### DISCUSSION TITLE
Programmatically refresh the route loader

### DISCUSSION THREAD

**Discussion by juangl** [remix-run/remix#1886] [Q&A] [answered]
I may be missing something obvious from the documentation but let's say I receive a webSocket notification and after that, I want to reload the data from a specific route, what's the best way to do it?

What I want is the same behavior as `unstable_shouldReload` + `<Form />`/`useFetcher`. Those APIs already reload the loaders, just want to do it programmatically without having to make a request.

**Comment by ccssmnn** [comment]
For reloading the current page, the browser has an [API](https://developer.mozilla.org/en-US/docs/Web/API/Location/reload): `window.location.reload()`.

The other use-case would be redirecting, which you could do with browser APIs as well or with React Routers [navigate](https://reactrouter.com/docs/en/v6/api#usenavigate)

**Reply by juangl** [reply]
this is not quite what I want. I edited my question to make it a little bit more specific.

**Comment by na2hiro** [comment, marked as answer]
Refresh API is proposed a few days ago. There's a current workaround in it too

https://github.com/remix-run/remix/discussions/1996
